### PR TITLE
Fix/lamport check and append circular buff in stigmergy

### PR DESCRIPTION
**Pull request by xgroleau**
This fixes  this [issue](https://github.com/MISTLab/BittyBuzz/issues/13), you can check it for more information.

When checked if a lamport is newer, if the lamports are equals, false is returned. This avoids always sending the same message back and forth between two hosts.

Also to add an item in the end, it's possible to use `bbzringbuf_rawat` instead. As of it's done now, when vm->outmsgs.queue.datastart is non zero, the value is written at the wrong index.

### CODE REVIEW

**Comment by beltrame** [comment]
When two lamports are equal, the vstig should call the conflict resolution mechanism (`bbzvstig_onconflict()`) so I think the problem might be in the conflict detection?

**Comment by xgroleau** [comment]
Just to be sure, the function bbzvstig_onconflict is to be called in the buzz script for user defined handling right? From my understanding, the conflict resolution happens [here](https://github.com/MISTLab/BittyBuzz/blob/cb791538d5c199409efef790f7439d2bedfc313f/src/bittybuzz/bbzmsg.c#L148) and calls the registered function, thus [bbzlamport_isnewer here](https://github.com/MISTLab/BittyBuzz/blob/cb791538d5c199409efef790f7439d2bedfc313f/src/bittybuzz/bbzmsg.c#L124) needs to return false to get there.


 In other words, from what I understand, the conflict  resolution is unreachable as of now without this fix since the bbzlamport_isnewer returns true when there is a conflict (when lamports are equal), and the newer lamport `if` statement is  before the conflict resolution.

**Comment by xgroleau** [comment]
I've tested it with my branch, and when the lamport are equals, the onconflict closure is called as expected. But never called when using the master since the message is considered "new" instead of a conflict when they are equal.

**Comment by beltrame** [comment]
You are correct, I'll accept the pull request.

**Comment by beltrame** [comment]
Thanks!
